PolyEngine/RenderingDevice/OpenGL/Src/GLWorldRendering.cpp

```diff
@@ -40,62 +38,171 @@ void GLRenderingDevice::RenderWorld(World* world)
+	else
+		PostprocessRenderingPasses[ePostprocessRenderPassType::BACKGROUND_LIGHT]->Run(world, cameraCmp, rect);
+
+	glDepthMask(GL_TRUE);
+	glEnable(GL_DEPTH_TEST);
+	// Render meshes with blin-phong shader
+	GeometryRenderingPasses[eGeometryRenderPassType::UNLIT]->Run(world, cameraCmp, rect);
+
+	glDepthMask(GL_FALSE);
+
+	glEnable(GL_BLEND);
+
+	// TODO test these blending options
+	glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+	//glBlendFunc(GL_ONE, GL_ONE);
+	//glBlendFunc(GL_ONE_MINUS_DST_COLOR, GL_ONE);
+
+	GeometryRenderingPasses[eGeometryRenderPassType::TRANSPARENT_GEOMETRY]->Run(world, cameraCmp, rect);
+
+	glDisable(GL_BLEND);
+
+	glDisable(GL_DEPTH_TEST);
+
+	// Run postprocess passes
+	// for (ePostprocessRenderPassType type : IterateEnum<ePostprocessRenderPassType>())
+
+	// Render text
+	GeometryRenderingPasses[eGeometryRenderPassType::TEXT_2D]->Run(world, cameraCmp, rect);
+
+	if (post && post->UseFgShader)
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND]->Run(world, cameraCmp, rect);
+	else
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND_LIGHT]->Run(world, cameraCmp, rect);
+
+	// Draw debug normals
+	if (gDebugConfig.DebugNormalsFlag)
+		GeometryRenderingPasses[eGeometryRenderPassType::DEBUG_NORMALS]->Run(world, cameraCmp, rect);
+}
+
+void GLRenderingDevice::RenderLit(World* world, const AARect& rect, CameraComponent* cameraCmp) const
+{
+	PostprocessSettingsComponent* post = cameraCmp->GetSibling<PostprocessSettingsComponent>();
+
+	glPolygonMode(GL_FRONT_AND_BACK, GL_FILL);
+	glDepthMask(GL_FALSE);
+	glDisable(GL_DEPTH_TEST);
+
+	if (post && post->UseBgShader)
+		PostprocessRenderingPasses[ePostprocessRenderPassType::BACKGROUND]->Run(world, cameraCmp, rect);
+	else
+		PostprocessRenderingPasses[ePostprocessRenderPassType::BACKGROUND_LIGHT]->Run(world, cameraCmp, rect);
+
+	glDepthMask(GL_TRUE);
+	glEnable(GL_DEPTH_TEST);
+	// Render meshes with blin-phong shader
+	GeometryRenderingPasses[eGeometryRenderPassType::BLINN_PHONG]->Run(world, cameraCmp, rect);
+
+	glDepthMask(GL_FALSE);
+
+	glEnable(GL_BLEND);
+
+	// TODO test these blending options
+	glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+	//glBlendFunc(GL_ONE, GL_ONE);
+	//glBlendFunc(GL_ONE_MINUS_DST_COLOR, GL_ONE);
+
+	GeometryRenderingPasses[eGeometryRenderPassType::TRANSPARENT_GEOMETRY]->Run(world, cameraCmp, rect);
+
+	glDisable(GL_BLEND);
+
+	glDisable(GL_DEPTH_TEST);
+
+	// Run postprocess passes
+	// for (ePostprocessRenderPassType type : IterateEnum<ePostprocessRenderPassType>())
+
+	// Render text
+	GeometryRenderingPasses[eGeometryRenderPassType::TEXT_2D]->Run(world, cameraCmp, rect);
+
+	if (post && post->UseFgShader)
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND]->Run(world, cameraCmp, rect);
+	else
+		PostprocessRenderingPasses[ePostprocessRenderPassType::FOREGROUND_LIGHT]->Run(world, cameraCmp, rect);
+
+	// Draw debug normals
+	if (gDebugConfig.DebugNormalsFlag)
```

Comment:
We should consider moving this to CameraComponent as well.

Reply:
My bad, this is already in camera component. Remove it from config then.

Reply:
Debug rendering modes are already in camera, now removed from config and removed these calls